persistence/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/AbstractPersistentEntityRegistry.scala

```diff
@@ -67,6 +67,10 @@ abstract class AbstractPersistentEntityRegistry(system: ActorSystem, injector: I
   }
 
   private val registeredTypeNames = new ConcurrentHashMap[String, Class[_]]()
+  private val reverseRegister: (Class[_]) => Option[String] = { entityClass =>
+    import scala.collection.JavaConversions.mapAsScalaConcurrentMap
+    registeredTypeNames.filter { case (name, className) => className == entityClass }.keys.headOption
+  }
```

Comment:
If anything, it's important for the class → name lookup to be really fast (since it happens all the time at runtime) and less important for the reverse lookup to be as fast (since registration should only really happen at startup time). IMO it would even be acceptable to do a linear search of values in `register` with `containsValue` and avoid the memory overhead of an extra map that's only used during startup and then retained forever.